### PR TITLE
Set dart runtime version with parsed `Platform.version`

### LINKED ISSUE #2141: Dart runtime contexts isn't easily queried
### Description

The current runtime contexts don't have a version field which makes it harder to query data, internally we depend on `version` so we can't currently retrieve that data.

**Current state**:

```json
"dart": {
  "name": "Dart",
  "raw_description": "3.4.3 (stable) (Tue Jun 4 19:51:39 2024 +0000) on \"android_x64\"",
  "type": "runtime"
},
```

**How it should be**

```json
"dart": {
  "name": "Dart",
  "version": "3.4.3 (stable)"
  "raw_description": "3.4.3 (stable) (Tue Jun 4 19:51:39 2024 +0000) on \"android_x64\"",
  "type": "runtime"
},
```

### PR DESCRIPTION
**Pull request by buenaflor**
## :scroll: Description

Fills the additional version field of the runtime context with version + channel

This makes it possible internally to query data based on dart runtimes

## :bulb: Motivation and Context

Closes #2141 

## :green_heart: How did you test it?
Unit test

## :pencil: Checklist

- [x] I reviewed submitted code
- [x] I added tests to verify changes
- [x] No new PII added or SDK only sends newly added PII if `sendDefaultPii` is enabled
- [ ] I updated the docs if needed
- [x] All tests passing
- [x] No breaking changes

### CODE REVIEW

File: dart/test/event_processor/enricher/io_enricher_test.dart [hunk 1]
```diff
@@ -25,6 +27,8 @@ void main() {
           .firstWhere((element) => element.name == 'Dart');
       expect(dartRuntime?.name, 'Dart');
       expect(dartRuntime?.rawDescription, isNotNull);
+      expect(dartRuntime!.version.toString() != Platform.version, true);
+      expect(Platform.version.contains(dartRuntime.version.toString()), true);
```
Comment (vaind) [inline dart/test/event_processor/enricher/io_enricher_test.dart]: Matchers provide better feedback on failures:
```suggestion
      expect(dartRuntime!.version.toString(), isNot(Platform.version));
      expect(Platform.version, contains(dartRuntime.version.toString()));
```

File: dart/lib/src/event_processor/enricher/io_enricher_event_processor.dart [hunk 2]
```diff
@@ -54,8 +54,21 @@ class IoEnricherEventProcessor implements EnricherEventProcessor {
   List<SentryRuntime> _getRuntimes(List<SentryRuntime>? runtimes) {
     // Pure Dart doesn't have specific runtimes per build mode
     // like Flutter: https://flutter.dev/docs/testing/build-modes
-    final dartRuntime = SentryRuntime(
+
+    // Extracts the semantic version and channel from the full version string.
+    // 3.5.0-180.3.beta (beta) (Wed Jun 5 15:06:15 2024 +0000) on "android_arm64"
+    // turns into 3.5.0-180.3.beta (beta)
+    String version = Platform.version;
+    RegExp channelRegex = RegExp(r'\((stable|beta|dev)\)');
```
Comment (vaind) [inline dart/lib/src/event_processor/enricher/io_enricher_event_processor.dart]: constructing & executing a RegExp may be relatively costly and this doesn't need to run on every event. Maybe consider extracting `dartRuntime` to a `late` variable.

Comment (buenaflor) [reply dart/lib/src/event_processor/enricher/io_enricher_event_processor.dart]: yup makes sense! thx

File: dart/lib/src/event_processor/enricher/io_enricher_event_processor.dart [hunk 3]
```diff
@@ -12,9 +12,26 @@ EnricherEventProcessor enricherEventProcessor(SentryOptions options) {
 /// Uses Darts [Platform](https://api.dart.dev/stable/dart-io/Platform-class.html)
 /// class to read information.
 class IoEnricherEventProcessor implements EnricherEventProcessor {
-  IoEnricherEventProcessor(this._options);
+  IoEnricherEventProcessor(this._options) {
+    dartVersion = _extractDartVersion(Platform.version);
+  }
 
   final SentryOptions _options;
+  late final String dartVersion;
```
Comment (vaind) [inline dart/lib/src/event_processor/enricher/io_enricher_event_processor.dart]: how about automatic init when needed? 
```suggestion
  IoEnricherEventProcessor(this._options);
  
  final SentryOptions _options;
  late final String dartVersion = _extractDartVersion(Platform.version);
```

Comment (vaind) [reply dart/lib/src/event_processor/enricher/io_enricher_event_processor.dart]: and maybe don't expose it as it's just an implementation detail

**Review by vaind — APPROVED** [review]

**Pull request merged**